Code/RDBoost/CMakeLists.txt

```diff
@@ -1,5 +1,12 @@
-rdkit_library(RDBoost Wrap.cpp
-              LINK_LIBRARIES RDGeneral ${PYTHON_LIBRARIES} ${Boost_LIBRARIES})
+if(APPLE)
```

Comment:
Note we should only do this for conda builds.  This will break system and brew based builds.

Reply:
and possible for conda above a specific version.  Perhaps we should simply make an RD_STATICALLY_LINKED_PYTHON flag or something like that.

Reply:
We may not need this patch.  If I use the following meta.yaml changes to boost:
```
    - boost ==1.61.0 [osx and py27]
    - boost ==1.61.0 [osx and py35]
    - boost ==1.65.1 [osx and py36]
```

and build via
```
conda build rdkit -c conda-forge
```

Apparently uses boosts that were built correctly.

This needs to be confirmed.

Reply:
I was under the impression that this was the recommended method for homebrew builds also. If I do a `brew audit rdkit` I get:

```
  * python modules have explicit framework links
    These python extension modules were linked directly to a Python
    framework binary. They should be linked with -undefined dynamic_lookup
    instead of -lpython or -framework Python.
      /usr/local/opt/rdkit/lib/python2.7/site-packages/rdkit/rdBase.so
            /usr/local/opt/rdkit/lib/python2.7/site-packages/rdkit/DataManip/Metric/rdMetricMatrixCalc.so
            /usr/local/opt/rdkit/lib/python2.7/site-packages/rdkit/DataStructs/cDataStructs.so
    [etc...]
Error: 1 problem in 1 formula
```

With regards, to boost, I found this PR from a couple of years ago when they switch to using `-undefined dynamic_lookup`: https://github.com/boostorg/build/pull/78. So I think all that's needed is a boost newer than that? The recent boost builds from conda-forge all do seem to be built correctly.

But it's worth noting that the conda-forge ecosystem doesn't (yet?) exhibit any issues with explicit linking against python, as they provide their own python build. So `conda install -c conda-forge rdkit` is indeed fine without this patch. It's those who are doing `conda install -c rdkit rdkit` and using the `defaults` python that are seeing issues.

Reply:
```conda build rdkit -c anaconda```
and 
```conda build rdkit -c conda-forge```

work correctly.  Using the defaults channel doesn't.  I'm going to try to get continuum to update boost and cairo on the defaults channel.

In the meantime, I would feel more comfortable setting this as an option as opposed to an osx default.  We still do builds on older conda installations and I don't know how this will affect it.